### ISSUE TITLE
Question: Free disk space check

### ISSUE THREAD

**Issue by jbruchanov** [rakshasa/rtorrent#784]
rTorrent 0.9.6/0.13.6
OS: Ubuntu 18.04.1 LTS (GNU/Linux 4.15.0-23-generic x86_64)

I can observer everytime crash when the rtorrent hits max limit of disk with quota.
I do have allocated space for 100GB, `df -h` is always more than my limit.
Is there a way how to set quota size for disk or some kind of maximum disk usage by rtorrent ?

**Comment by chros73** [comment]
> Is there a way how to set quota size for disk or some kind of maximum disk usage by rtorrent ?

Not really. Although there's a directive that suppose to handle this, but it doesn't work reliably in 0.9.6:
```ini
# Close torrents when diskspace is low.
schedule2 = low_diskspace, 60, 60, close_low_diskspace=1000M
```

You need proper queue management:
- [pyrotorque](https://pyrocore.readthedocs.io/en/latest/advanced.html?highlight=pyrotorque#rtorrent-queue-manager)
- or as this [setup](https://github.com/chros73/rtorrent-ps-ch_setup) uses

**Comment by jbruchanov** [comment]
I don't think that's what I'm looking for.

It's a 30TB share drive for multiple users, so I can't really rely on free space on the drive, because it doesn't reflect my used quota.
I need something what is used based on actual space taken by data in my folder...

Or just fix the crashing issue when it is unable to write data even if there is "enough" space on the drive

**Comment by chros73** [comment]
Hmm...
So, are you sure that `close_low_diskspace=1000M` command doesn't work in your case?

If so, then you can script it somehow like this:
- only load all the torrents but not start them with `load.normal` command (instead of using `load.start`)
  - only start them if there is space for it/them by using an external script that checks it

**Comment by kannibalox** [comment]
No offense, but I think you're misunderestimating what pyrotorque can do. Modify [this section](https://pyrocore.readthedocs.io/en/latest/custom.html#only-start-items-that-you-have-disk-space-for) to check for free space on your as-yet-unspecified quota tooling of choice, and you're good to go.

**Comment by jbruchanov** [comment]
I'll check the 'pyrotorque' my answer was for the `close_low_diskspace` which if I understood docs correctly checking free disk space and that won't help me, because my quota can be full, but disk can still have many TBs free.

Anyway I believe as the rtorrent is crashing in this case, it's still valid bug.

**Comment by chros73** [comment]
> Anyway I believe as the rtorrent is crashing in this case, it's still valid bug.

That's correct.

> ... close_low_diskspace which if I understood docs correctly ...

Don't rely on the docs, quickly try out that command:
- run rtorrent
- hit `^x` and enter: `close_low_diskspace=1000M` (use a value that will be over your current quota)
  - you should see that torrents are closing
